spec/requests/api/move_events_controller_redirect_spec.rb

```diff
@@ -108,18 +79,68 @@
 
     context 'with validation errors' do
       context 'with a bad timestamp' do
-        let(:redirect_params) { { data: { type: 'redirects', attributes: { timestamp: 'Foo-Bar' } } } }
+        let(:redirect_params) do
+          { data: { type: 'redirects',
+                    attributes: { timestamp: 'Foo-Bar' },
+                    relationships: {
+                      to_location: { data: { type: 'locations', id: new_location.id } },
+                    } } }
+        end
 
         it_behaves_like 'an endpoint that responds with error 422' do
-          let(:errors_422) { [{ 'title' => 'Invalid timestamp', 'detail' => 'Validation failed: Timestamp must be formatted as a valid ISO-8601 date-time' }] }
+          let(:errors_422) do
+            [{
+              'title' => 'Invalid timestamp',
+              'detail' => 'Validation failed: Timestamp must be formatted as a valid ISO-8601 date-time',
+            }]
+          end
         end
       end
 
       context 'with a bad event type' do
         let(:redirect_params) { { data: { type: 'Foo-bar', attributes: { timestamp: '2020-04-23T18:25:43.511Z' } } } }
 
         it_behaves_like 'an endpoint that responds with error 422' do
-          let(:errors_422) { [{ 'title' => 'Invalid type', 'detail' => 'Validation failed: Type is not included in the list' }] }
+          let(:errors_422) do
+            [{
+              'title' => 'Invalid type',
+              'detail' => 'Validation failed: Type is not included in the list',
+            }]
+          end
+        end
+      end
+
+      context 'with a missing to_location relationship' do
+        let(:redirect_params) { { data: { type: 'redirects', attributes: { timestamp: '2020-04-23T18:25:43.511Z' } } } }
+
+        it_behaves_like 'an endpoint that responds with error 422' do
```

Comment:
it sounds we should should remove validation tests from here

Reply:
I have removed a validation test and am checking in the params validator instead